### PR TITLE
plugins/colorschemes: migrate helpers -> lib.nixvim

### PR DESCRIPTION
**Pull request by khaneliman**
I see all the colorschemes are like this, gonna just update them all here

### CODE REVIEW

**Comment by MattSturgeon** [comment]
> gonna just update them all here

Yeah these kinda migrations are fine to be done in bulk. Hopefully less painful than #1674 :crossed_fingers: 

There's no rush to get this migration work done though, it's less of a priority than migrating old plugins to `mkNeovimPlugin`.

File: plugins/colorschemes/cyberdream.nix [hunk 1]
```diff
@@ -1,42 +1,41 @@
...
   isColorscheme = true;
   originalName = "cyberdream.nvim";
   defaultPackage = pkgs.vimPlugins.cyberdream-nvim;
 
-  maintainers = [ helpers.maintainers.AndresBermeoMarinelli ];
+  maintainers = [ lib.nixvim.maintainers.AndresBermeoMarinelli ];
 
   settingsOptions = {
-    transparent = helpers.defaultNullOpts.mkBool false ''
+    transparent = defaultNullOpts.mkBool false ''
```
Comment (MattSturgeon) [inline plugins/colorschemes/cyberdream.nix:17]:
```
error: undefined variable 'defaultNullOpts'
at /nix/store/XXX-source/plugins/colorschemes/cyberdream.nix:29:28:
```

**Comment by khaneliman** [comment]
> > gonna just update them all here
> 
> Yeah these kinda migrations are fine to be done in bulk. Hopefully less painful than #1674 🤞
> 
> There's no rush to get this migration work done though, it's less of a priority than migrating old plugins to `mkNeovimPlugin`.

Yeah, was just doing some of them out of OCD and they are quicker victories than migrating an entire plugins options to a new way of doing things. :P

File: plugins/colorschemes/ayu.nix [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
-  helpers,
   config,
   pkgs,
   ...
 }:
 with lib;
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+  inherit (lib.nixvim) defaultNullOpts toLuaObject;
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/ayu.nix:10]:
```suggestion
```

File: plugins/colorschemes/catppuccin.nix [hunk 3]
```diff
@@ -1,12 +1,15 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
 with lib;
```
Comment (MattSturgeon) [inline plugins/colorschemes/catppuccin.nix:7]: (All?) the files still have `with lib`

File: plugins/colorschemes/catppuccin.nix [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+  inherit (lib.nixvim) defaultNullOpts mkNullOrOption mkNullOrStrLuaFnOr;
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/catppuccin.nix:10]:
```suggestion
```

File: plugins/colorschemes/base16/default.nix [hunk 5]
```diff
@@ -10,8 +9,10 @@ let
   name = "base16";
   luaName = "base16-colorscheme";
   originalName = "base16.nvim";
+
+  inherit (lib.nixvim) defaultNullOpts toLuaObject;
```
Comment (MattSturgeon) [inline plugins/colorschemes/base16/default.nix:13]: Maybe better at the top of the let block?

File: plugins/colorschemes/cyberdream.nix [hunk 6]
```diff
@@ -1,42 +1,45 @@
...
   lib,
-  helpers,
   pkgs,
   ...
 }:
 with lib;
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+  inherit (lib.nixvim) defaultNullOpts;
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/cyberdream.nix:10]:
```suggestion
```

File: plugins/colorschemes/nightfox.nix [hunk 7]
```diff
@@ -1,12 +1,15 @@
...
   lib,
-  helpers,
   pkgs,
   config,
   ...
 }:
 with lib;
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/nightfox.nix:9]:
```suggestion
```

File: plugins/colorschemes/rose-pine.nix [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
-  helpers,
   pkgs,
   config,
   ...
 }:
 with lib;
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+  inherit (lib.nixvim) defaultNullOpts mkNullOrOption;
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/rose-pine.nix:10]:
```suggestion
```

File: plugins/colorschemes/tokyonight.nix [hunk 9]
```diff
@@ -1,12 +1,15 @@
...
-  helpers,
   pkgs,
   config,
   ...
 }:
 with lib;
-helpers.neovim-plugin.mkNeovimPlugin config {
+let
+  inherit (lib.nixvim) defaultNullOpts mkNullOrOption;
+
```
Comment (MattSturgeon) [inline plugins/colorschemes/tokyonight.nix:10]:
```suggestion
```

File: plugins/colorschemes/catppuccin.nix (hunk 3 above)
Comment (MattSturgeon) [reply plugins/colorschemes/catppuccin.nix:7]: IDK if you'd rather keep that out of scope for this PR though?

**Review by MattSturgeon — CHANGES_REQUESTED** [review]

File: plugins/colorschemes/catppuccin.nix (hunk 3 above)
Comment (khaneliman) [reply plugins/colorschemes/catppuccin.nix:7]: I can add another commit for that

**Review by MattSturgeon — APPROVED** [review]

File: plugins/colorschemes/catppuccin.nix
Comment (MattSturgeon) [inline plugins/colorschemes/catppuccin.nix:1]: This file's diff would be much smaller with an `inherit (lib) types`

File: plugins/colorschemes/kanagawa.nix
Comment (MattSturgeon) [inline plugins/colorschemes/kanagawa.nix:1]: Same (`inherit (lib) types`)

File: plugins/colorschemes/nightfox.nix
Comment (MattSturgeon) [inline plugins/colorschemes/nightfox.nix:1]: Same

**Comment by khaneliman** [comment]
Sounds good, wasn't sure about that one. But, it does make it cleaner and still shows ownership of where it's coming from.

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**